crates/astria-conductor/src/data_availability.rs

```diff
@@ -0,0 +1,382 @@
+                 should have been received; this is a bug"
+            );
+            return;
+        }
+        self.process_sequencer_datas.spawn(
+            height,
+            process_sequencer_data(
+                height,
+                sequencer_data,
+                self.celestia_client.clone(),
+                self.block_verifier.clone(),
+                self.namespace,
+            )
+            .in_current_span(),
+        );
+    }
+
+    #[instrument(skip_all, fields(height))]
+    fn send_sequencer_subsets(
+        &self,
+        sequencer_subsets_res: Result<eyre::Result<Vec<SequencerBlockSubset>>, JoinError>,
+    ) -> eyre::Result<()> {
+        let subsets = match sequencer_subsets_res {
+            Err(e) => {
+                warn!(error.message = %e, error.cause = ?e, "task processing sequencer data failed");
+                return Ok(());
+            }
+            Ok(Err(e)) => {
+                warn!(error.message = %e, error.cause = ?e, "task processing sequencer data returned with an error");
+                return Ok(());
+            }
+            Ok(Ok(subsets)) => subsets,
+        };
+        self.executor_tx
+            .send(executor::ExecutorCommand::FromCelestia(subsets))
+            .wrap_err("failed sending processed sequencer subsets: executor channel is closed")
+    }
+}
+
+async fn process_sequencer_data(
+    height: u64,
+    datas: Vec<SignedNamespaceData<SequencerNamespaceData>>,
+    client: CelestiaClient,
+    block_verifier: BlockVerifier,
+    namespace: Namespace,
+) -> eyre::Result<Vec<SequencerBlockSubset>> {
+    // spawn the verification tasks
+    let mut verification_tasks = verify_all_datas(datas, block_verifier);
+
+    let (assembly_tx, assembly_rx) = mpsc::channel(256);
+    let block_assembler = task::spawn(assemble_blocks(assembly_rx));
+
+    let mut get_rollups = JoinSet::new();
+    while let Some((block_hash, verification_result)) = verification_tasks.join_next().await {
+        match verification_result {
+            Err(e) => {
+                warn!(%block_hash, error.message = %e, error.cause = ?e, "task verifying sequencer data retrieved from celestia failed; dropping block")
+            }
+            Ok(Err(e)) => {
+                warn!(%block_hash, error.message = %e, error.cause = ?e, "task verifying sequencer data retrieved from celestia returned with an error; dropping block")
+            }
+            Ok(Ok(data)) => {
+                get_rollups.spawn(
+                    get_rollup(client.clone(), height, data, namespace, assembly_tx.clone())
+                        .in_current_span(),
+                );
+            }
+        }
+    }
+
+    // ensure that the last sender goes out of scope so that block assembler's exit condition fires
+    std::mem::drop(assembly_tx);
+
+    block_assembler
+        .await
+        .wrap_err("failed assembling sequencer block subsets")
+}
+
+async fn get_rollup(
```

Comment:
```suggestion
async fn get_rollup_data(
```

Reply:
Changed this to `fetch_and_verify_rollup_blobs`